lib/run_loop/plist_buddy.rb

```diff
@@ -26,6 +26,16 @@ def plist_read(key, file, opts={})
       end
     end
 
+    # !!Experimental!! find device in xcuserstate file.
+    # @param [String] file the plist to read
+    # @return [String] the UDID of device
+    def plist_find_device(file)
+      `#{plist_buddy} -c Print "#{file}"`
+          .encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
+          .split("\n")
+          .select { |v| v =~ /dvtdevice.*:/ }[0]
+    end
+
```

Comment:
>I wish we could add something like this to PlistBuddy, but this class it not well written which >means it is difficult to extend.

I have tried to add something like `simple read` method but for me it looks like existing running `pbuddy` command mechanism hangs on big output and in this case it just huge. But if you have an idea how to handle that issue i can rewrite/update implementation. It is not critical functionality and we have time to make in right way.